improver_tests/ensemble_copula_coupling/test_utilities.py

```diff
@@ -398,5 +402,59 @@ def test_multiple_timesteps(self):
         self.assertArrayAlmostEqual(reshaped_array, expected)
 
 
+numba_installed = True
+try:
+    importlib.util.find_spec("numba")
+    from improver.ensemble_copula_coupling.numba_utilities import fast_interp_same_x
+except ImportError:
+    numba_installed = False
+
+
+class TestInterpolateMultipleRowsSameX(IrisTest):
+
+    """Test interpolate_multiple_rows"""
+
+    def setUp(self):
+        """Set up arrays."""
+        np.random.seed(0)
+        self.x = np.arange(0, 1, 0.01)
+        self.xp = np.sort(np.random.random_sample(100))
+        self.fp = np.random.random((100, 100))
+
+    @skipIf(not (numba_installed), "numba not installed")
+    def test_slow_vs_fast(self):
+        """Test that slow and fast versions give same result."""
+        result_slow = slow_interp_same_x(self.x, self.xp, self.fp)
+        result_fast = fast_interp_same_x(self.x, self.xp, self.fp)
+        np.testing.assert_allclose(result_slow, result_fast)
+
+    @skipIf(not (numba_installed), "numba not installed")
+    def test_slow_vs_fast_unordered(self):
+        """Test that slow and fast versions give same result
+        when x is not sorted."""
+        shuffled_x = self.x.copy()
+        np.random.shuffle(shuffled_x)
+        result_slow = slow_interp_same_x(shuffled_x, self.xp, self.fp)
+        result_fast = fast_interp_same_x(shuffled_x, self.xp, self.fp)
+        np.testing.assert_allclose(result_slow, result_fast)
+
+    @skipIf(not (numba_installed), "numba not installed")
+    def test_slow_vs_fast_repeated(self):
+        """Test that slow and fast versions give same result when xp
+        contains repeats."""
+        repeat_xp = self.xp.copy()
+        repeat_xp[51] = repeat_xp[50]
+        result_slow = slow_interp_same_x(self.x, repeat_xp, self.fp)
+        result_fast = fast_interp_same_x(self.x, repeat_xp, self.fp)
+        np.testing.assert_allclose(result_slow, result_fast)
+
+    def test_slow_vs_multi(self):
+        """Test that slow interp gives same result as
+        interpolate_multiple_rows_same_x."""
+        result_slow = slow_interp_same_x(self.x, self.xp, self.fp)
+        result_multiple = interpolate_multiple_rows_same_x(self.x, self.xp, self.fp)
+        np.testing.assert_allclose(result_slow, result_multiple)
+
+
```

Comment:
There are no tests in here with a known result. I'm happy with the use of random numbers for the comparison tests, but can you please add a simple case with a known set of x, xp, fp, and a result to demonstrate that the methods return what we would get with paper and pencil. Thanks.

Reply:
Done